app/src/main/java/com/example/android/tvleanback/ui/PlaybackOverlayFragment.java

```diff
@@ -505,4 +481,40 @@ public void onItemClicked(Presenter.ViewHolder itemViewHolder, Object item,
         }
     }
 
+    private class MediaControllerCallback extends MediaController.Callback {
+        @Override
+        public void onPlaybackStateChanged(PlaybackState state) {
+            Log.d(TAG, "playback state changed: " + state.getState());
+            if (state.getState() == PlaybackState.STATE_PLAYING && mCurrentPlaybackState != PlaybackState.STATE_PLAYING) {
```

Comment:
I'm worried that playback UI could get out of sync, since we're only checking for changes in known state variables. For example, `onStop()` calls `stopProgressAutomation()`, so `startProgressAutomation()` might not be called correctly in this method. Similarly, it's hard to determine if other UI state could get out of sync.

Is there a performance reason to check `mCurrentPlaybackState`? If not, I would just check `state.getState()` to make the code more readable.

`if (state.getState() == PlaybackState.STATE_PLAYING) {`
`    // Set Play state`
`} else { // else, not else if`
`    // Set Pause state`
`}`